### ISSUE TITLE
[Suggestion] Random Loot Chests

### ISSUE THREAD

**Issue by Shaners** [HabitRPG/habitica#550]
From a personal user experience perspective Habit RPG has been motivating and fun but after a month of use I feel my interest waning a bit and things becoming monotonous. Every time I complete something I see the same gold and experience pop up and be added to my pool and nothing seems to change. 

My suggestion would be that there would be a 3 - 5% chance each time the player received a reward for it to instead be a loot reward chest that would be clicked on, had an animation of the chest opening and / or glowing, and gave the player a random assortment of items. The benefits to the player would be 3 fold: broken up monotony, cool animation to watch, and the actual items in the chest. The chance to get a chest could obviously be altered but you wouldn't want it to be too prevalent so as to be annoying and just replace one monotony with another or too sparse that a very low percentage of players see it. As a middle ground you could introduce tiered chests (wood chest, water logged chest, pirate chest, ancient chest, iron chest, gold chest, pearl chest etc.) where the lower tiers were more common but had lesser rewards.

I'm sure this has been suggested before but I wasn't really sure where I could add to that conversation and I wanted to stress that I think it would be a good idea to consider this a priority task at this stage in Habit RPG's development. It may be reliant on other systems being build (an inventory system) but there must be someway to get the bare bones in.

Some possible items you might see in a chest:
- Cheese (Food items could restore HP or give temporary stat boosts)
- Fresh Milk
- Potato
- Slice of Watermelon
- Honey
- Chocolate
- Ancient Tome ( starts a quest? )
- Gold ( randomly more or less than what you would have normally got )
- Equipment ( lowest tier chests only have lower gear )

**Comment by Pandoro** [comment]
This remembers me of the second idea I had in #199 (Second post)

Maybe you would want to check that out as well. I proposed quests that
require you to gather special drops from tasks. This pretty much boils down
to the same principle, you randomly get stuff from tasks. I just think that
a quest would make it even a bit more fun, considering you don't just get
something, but you need to work to collect several things.

I never thought of chests though, which make the whole thing even cooler.
The question is would we want to do chest -> has chance to start quest or
rather quest -> has chance of chest? Or both in parallel?

I really like the randomness in your proposal, as it will keep you engaged,
hoping you get this one rare thing out of a chest. Personally I would also
love a very rare chance for a pet in these scenarios. Of course it goes a
little against the monetization idea, but the hope of getting one would be
very cool for people who don't have the money! OR alternatively, you could
get pet customizations out of these quests, like a zombie elixir, a flesh
eating poison, candy cotton stuffing etc.  which could modify your already
bought pets.

There are so many options...

**Comment by Shaners** [comment]
Definitely think they could work in parallel and compliment each other. 
- Every completed task has a small chance to drop a chest instead of the regular rewards
- Of those rare loot chests there is a small chance to get an item that starts a quest (ancient scroll, dwarven ruby etc.) that may or may not be available by other means
- Quests can be built by the user that represent more complicated or multifaceted projects with stages that they need to get done in real life
- Quests can be offered by the town caller that are tied to 'in game' events or lore where progress is made by doing any old tasks you already have
- IF a quest is accepted or activated by user have the chance of chests dropping increased the whole duration as a reward (maybe have hp loss increased during this time as well to help balance it?)
- 100% chance for a chest reward after completing a quest

This may encourage people to take on projects they have been putting off if they have a digital counter-part mirroring and in essence encouraging them along the way.

I also really like the idea of having the pet modification items in the chests too. Making it rare enough still saves the shop since it'll be very hard to get the specific one you are looking for but doesn't make it impossible for those users for whom buying tokens may not be an option (younger children?) You could also make some of the premium super rare or token only purchases (e.g. gold pets have to be bought.)

**Comment by Pandoro** [comment]
I would like to comment/ discuss each of your points @Shaners :) (Already
sorry for the long text :x)
1. How do you mean instead of the regular reward? Do you mean the EXP+GP or
   do you mean if quests were there, the normal loot would be exchanged by a
   chest?
2. Yes, I think these two systems can both build on top of each other,
   synergy ftw! But I would kind of limit the chain. It would suck if you get
   a quest, which drops a chest, which gives you a quest, which drops a chest,
   which gives ... you know ;) Furthermore my original idea of quests would
   actually specify the reward you get. As with the dwarven ring example. It
   would kind of suck if someone gets this quest, because he/she was looking
   for this ring since a long time and then a chest would drop. So I would
   make it clear what the reward would be. Either completely random, or a
   fixed item, or something like 15% change of item B, 1% chance of super rare
   item C and a 84% chance of a chest.
3. See my proposal at the end of this post.
4. Very much like the idea. I have to suppress a tear of nostalgia when
   thinking about the FFXI events right now... Having something like this
   would REALLY be cool. Imagine how holidays could become the most productive
   times in our lives, because we REALLY WANT THAT PUMPKIN HAT! ;)
5. Possible, possible! I do feel a little though, as if these quests should
   be a little "rare". I don't like the idea of having 4 quests active all the
   time. This would shift the whole focus of HabitRPG from doing your tasks,
   to a quest based system.
6. Hmmm this conflicts a little to point 2 and maybe 1? Please correct me
   if I am wrong.

Regarding point 3. This is what I could image.
Say you have different quest level frameworks, easy, medium and hard. Given
one framework you would have to place certain tasks, todos or dailies into
the given slots and you would need to complete these in order to complete
the quest?

For example: (Wild ideas here..., feel free to tell me "NOPE!")
Easy:
1 Boss slot
5 mid stage slots
3 starting slots
Here I would imagine some graph, where each mid stage slot is linked to 2
starting slots. These you would have to complete in order to complete the
mid stage slots. The boss slot is linked to all mid stage slots, meaning
you will need to finish all mid stage, and thus also all starting stage
slots in order to do the boss slot. Completing would yield you some fancy
reward. Here completing does not mean you can't complete the actual todo,
but you cannot finish the quest until you have done the requested links.
The problem is that this pretty much just reduces it to a list of things
you need to complete. So I would suggest some added penalty/reward based on
the completion order for  tasks in a quest.

Example.
Boss: Finish seminar paper (Proof read and hand in)
Mid: Finish chapter 1,   2,   3,  4,  finish references.
Start: Setup template, search for reference papers, finish outline.

Medium
1 BOSS slot
2 sub-boss slots
5 mid stage slots connected to sub-boss 1 and 5 mid stage slots connected
to sub-boss 2
7 starting slots
Same kind of stuff like the easy one.

Pros of this:
+you would be able to get a stronger feeling of progress.
+you would get a better incentive to do stuff, considering you are working
to some random, possibly magical awesome, reward :)
- fun fun fun
  Cons
- A lot of implementation ;)
- The different framework links would probably need to be static, but might
  not fit MANY actual real life counterpart quests.
- People might start to put in random habits and clicking them just to get
  rewards. (This could be fixed by having certain requirements to the quests,
  like you cannot add stuff which you are very good at already, or it would
  have to be purely todos.)

Next to this, I would really like story based quests though, where we can
integrate some kind of "history and lore" of HabitRPG.

**Comment by lefnire** [comment]
FWIW, we've just begun building the "drop system" - small % loot drop. We're starting it for meat and eggs in the new [pet system](https://trello.com/card/pets-and-mounts/50e5d3684fe3a7266b0036d6/166). Will later drop weapon / armor [enchantments](https://trello.com/card/armor-weapon-enchantments/50e5d3684fe3a7266b0036d6/202). More items will be added over time, so count on this feature. I also really like the idea of dropped quest scrolls.

**Comment by Shaners** [comment]
@lefnire looks good. I'm going to have to learn how to use Trello (seems pretty straight forward) so I can put my comments on there. Are we just going to go for a higher res for icon assets? The gold and silver icons are already there so it probably isn't so much of a leap.

@Pandoro 

> How do you mean instead of the regular reward? Do you mean the EXP+GP or
> do you mean if quests were there, the normal loot would be exchanged by a
> chest?

Just meant when you complete a task right now you get 1 GP 1 XP. If the chest reward happened (~3% of the time) it could replace the XP and GP reward for that instance. Or the user could receive both for that instance, was just thinking of balance a bit.

> It would suck if you get
> a quest, which drops a chest, which gives you a quest, which drops a chest

Agreed, that is the function of the rarity. If drops and getting a quest from a drop weren't rare then you are just replacing monotony with a different type of monotony. (Monotony seems a bit hash as a term here though.) If you have a 1 in 100 chance of getting a chest and a 1 in 5 chance of a chest containing an item that leads to a quest the average user will have to complete 500 tasks to get an item from a chest that leads to a quest. 

> Very much like the idea. I have to suppress a tear of nostalgia when
> thinking about the FFXI events right now

This was something I wanted to suggest as well, global events where all the users could work towards defeating some giant god monster or something. May encourage rampant cheating though so still needs to be fleshed out a bit.

> I do feel a little though, as if these quests should
> be a little "rare". I don't like the idea of having 4 quests active all the
> time. 

Good point, I still wouldn't want users restricted to how many they can create for themselves though.

> Hmmm this conflicts a little to point 2 and maybe 1? Please correct me
> if I am wrong.

Ah to clarify what I meant was after a quest was completed successfully that you were guaranteed a chest as a reward. Would encourage quest competition and help people that were unlucky to not get one through the duration of the quest. <-- Would make less sense if getting chests were a requirement to completing a quest.

> For example: (Wild ideas here..., feel free to tell me "NOPE!")

I like this and it follows a similar path to what I was thinking. My only concern would be that it has to be easy for the user to set it up or they won't do it. Having requirements for slot types might be an impediment in that sense. Maybe the easy / basic level can just have a min number of tasks requirement?

**Comment by Shaners** [comment]
Going to close this since I think it's covered by the drop system feature.

**Comment by Pandoro** [comment]
Just as a general comment. The drop percentage should somehow be normalized
by the number of tasks people have. If someone has 60 dailies, they should
not get 10 times the amount of drops and quests as someone with only 6
dailies.

@Shaners regarding the guaranteed drop of a chest after a quest. I was
asking, because I was wondering how this would tie in with my quest
proposal. (Think about the dwarven ring quest) You would get a predefined
reward, like a fancy item. Would you say, that they also always get a
chest? Or did you mean this as an reward for the user defined quests?

**Comment by Shaners** [comment]
@pandoro you could just have the dwarven ring inside a chest (dwarven chest?) with no random reward or a lesser random reward. Alternatively if the ring had to be inside an old scroll or  something like that to fit with the quest's storyline you could skip it for that one. Really the chest is just a mechanic to display rewards whether it be random loot, quest-end reward, or bought from the shop.

**Comment by Pandoro** [comment]
Aight! Then we agree :) I really like the idea of having the final reward
in a chest, I think we should have both random quests, where the chest
contains something random and we should have story line quests, where you
could get the story line item (95%) or a rare version of the story line
item (5%) which for example has a little better gold boost or whatever
would make sense :)

**Comment by wc8** [comment]
:+1: for both random quests
:+1: for normalization
Towards normalization, perhaps, it could be based on the number of days you've been working at it? For example, if Period 2 is Day 14 through 21 in, on some random day between day 14 and day 21, some random task opens a quest. Just a thought.

**Comment by Pandoro** [comment]
Which period do you mean in this case @wc8?

**Comment by wc8** [comment]
I meant a period of time.
At first I thought, a quest after 3 days, a quest after 12 days, but then I thought, that's hardly random.
So then I thought, what if the first one came somewhere between day 5 and day 13, and the next one, some where between day 14 and day 21. But I'm beginning to think it's either a bad idea or a bad way of expressing it.

**Comment by Pandoro** [comment]
Hmmmm, making it time based is not a bad idea I think. That way you would
get the same amount regardless of your speed. You would just have fiddle
out a good way of implementing this.

Honestly I would like to have quests more often though, but that is a
different issue again. Although if quests are able to overlap, this time
based approach is fairly hard. So maybe we should keep some statistics of
how often people click stuff each day on average and then based on that
statistic give them random chances to obtain 1 quest at least every X days
(which translates to X*"average completion rate" finished tasks.)

**Comment by wc8** [comment]
My example time periods were just to explain the concept, not meant as a frequency suggestion. Since this idea seems to be a good incentive to keep checking and working off one's to-do list, I don't think it should be too rare. The particular example I gave is probably not good, frequency wise.

**Comment by Pandoro** [comment]
Ok :) Understood.  Maybe my idea is actually not so good either. As the
average will always converge to some value, meaning if I push it very hard
all the time, this becomes the average and thus I need to push it very hard
to actually get just a normal chance for a quest.

Maybe we will need to normalize based on dailies really in this case, then
again this might conflict with other peoples' playing style...

**Comment by Shaners** [comment]
You could track how many quests the player has been offered and how many they have completed and scale the chance for quest based on that with the goal of getting the player at least one quest a week. You could also monitor how many times the player declines (if declining is an option, it should be) to accept a quest and scale the chance for random quest back, say to 1 per 2 weeks since in that case you might assume the player does not want to do quests.

**Comment by Pandoro** [comment]
Yes Yes, good thinking!

EDIT: Although we have to occasionally query users who have declined many quests. This could otherwise set the chance so small that people would never get any quests at some point.
